packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -41,17 +42,32 @@ const bYield = (expr: EsExpression) => b.expressionStatement(b.yieldExpression(e
 // TODO [#4714]: scope token renders as a suffix for literals, but prefix for expressions
 const bConditionalLiveYield = esTemplateWithYield`
     {
-        const shouldRenderScopeToken = ${/* isClass */ is.literal} &&
+        const attrName = ${/* attribute name */ is.literal};
+        let attrValue = ${/* attribute value expression */ is.expression};
+        const isHtmlBooleanAttr = ${/* isHtmlBooleanAttr */ is.literal};
+
+        const shouldRenderScopeToken = attrName === 'class' &&
             (hasScopedStylesheets || hasScopedStaticStylesheets(Cmp));
         const prefix = shouldRenderScopeToken ? stylesheetScopeToken + ' ' : '';
 
-        const attrValue = ${/* attribute value expression */ is.expression};
-        const valueType = typeof attrValue;
+        // Global HTML boolean attributes are specially coerced into booleans
+        // https://github.com/salesforce/lwc/blob/f34a347/packages/%40lwc/template-compiler/src/codegen/index.ts#L450-L454
+        if (isHtmlBooleanAttr) {
+            attrValue = attrValue ? '' : undefined;
+        }
+
+        // Global HTML "tabindex" attribute is specially massaged into a stringified number
+        // This follows the historical behavior in api.ts:
+        // https://github.com/salesforce/lwc/blob/f34a347/packages/%40lwc/engine-core/src/framework/api.ts#L193-L211
+        if (attrName === 'tabindex') {
+            const shouldNormalize = attrValue > 0 && !(attrValue === true || attrValue === false);
```

Comment:
```suggestion
            const shouldNormalize = attrValue > 0 && attrValue !== true;
```
`false > 0` is already false, so the second check isn't necessary. (`true > 0` is true, though, so that extra check is needed. Thanks, JavaScript. 😒)

Reply:
```suggestion
            const shouldNormalize = attrValue > 0 && typeof attrValue !== 'boolean';
```
An alternative option that makes intent clear for both `true` and `false`, but without adding a strictly unnecessary check.